content/en/blog/2023/network-security-splunk/index.md

```diff
@@ -0,0 +1,70 @@
+
+## Securing Layer 7: Istio
+
+Splunk uses Istio to enforce policy on the application layer based on the details of each request. Istio also emits Telemetry data (metrics, logs, traces) that are useful for validating request-level security.
+
+One of the key benefits of all those sidecars is that Istio can provide in-transit encryption for the entire mesh without requiring any modifications to the applications. The applications send plain text HTTP requests, but the Envoy sidecar intercepts the traffic and implements Mutual TLS encryption to protect against interception or modification.
+
+Istio manages Splunk’s ingress gateways, which receive traffic from public and internal NLBs.  The gateways are managed by the platform team and run in the Istio Gateway namespace, allowing users to plug into them, but not modify them. The Gateway service is also provisioned with certificates to enforce TLS by default, and Validating Webhooks ensure that services can only connect to gateways for their own hostnames.  Additionally, gateways enforce request authentication at ingress, before traffic is able to impact application pods.
+
+Because Istio and related K8S objects are relatively complex to configure, Splunk created an abstraction layer, which is a controller that configures everything for the service, including virtual services, destination rules, gateways, certificates, and more. It sets up DNS that goes directly to the right NLB. It's a one-click solution for end-to-end network deployment.  For more complex use cases, the services teams can still bypass the abstraction and configure these settings directly.
+
+{{< image width="90%"
+    link="Splunk Platform.png"
+    caption="Splunk Application Platform"
+    >}}
+
+## Pain Points
+
+While Splunk’s architecture meets many of our needs, there are a few pain points worth discussing.  As you recall, Istio operates by creating as many Envoy Sidecars as application pods, which is an inefficient use of resources.  In addition, when a particular application has unique needs from its sidecar, such as additional CPU or Memory, it can be difficult to adjust these settings without adjusting them for all sidecars in the mesh.  Istio Sidecar injection involves a lot of magic, using a mutating webhook to add a sidecar container to every pod as it is created, which means those pods no longer match their corresponding deployments.  Additionally, injection can only happen at pod creation time, which means that any time a sidecar version or parameter is updated, all pods must be restarted before they will get the new settings.  Overall, this magic complicates running a service mesh in production, and adds a great deal of operational uncertainty to your application.
```

Comment:
```suggestion
While Splunk’s architecture meets many of our needs, there are a few pain points worth discussing.  Istio operates by creating as many Envoy Sidecars as application pods, which is an inefficient use of resources.  In addition, when a particular application has unique needs from its sidecar, such as additional CPU or Memory, it can be difficult to adjust these settings without adjusting them for all sidecars in the mesh.  Istio Sidecar injection involves a lot of magic, using a mutating webhook to add a sidecar container to every pod as it is created, which means those pods no longer match their corresponding deployments.  Additionally, injection can only happen at pod creation time, which means that any time a sidecar version or parameter is updated, all pods must be restarted before they will get the new settings.  Overall, this magic complicates running a service mesh in production, and adds a great deal of operational uncertainty to your application.
```